edx_repo_tools/repo_access_scraper/repo_access_scraper.py

```diff
@@ -0,0 +1,175 @@
+        screenshot_pages(page, url, f"access-{file_slug(repo)}")
+
+        # Get the team memberships
+        url = f"https://api.github.com/repos/{repo}/teams?per_page=100"
+        teams = request_list(url)
+        for team in teams:
+            perm = PERMS.index(team["permission"])
+            if perm < PUSH:
+                # We don't care about triage or pull access.
+                continue
+
+            name = team["name"]
+            team_page = team["html_url"]
+            access = ACCESS_NAMES[perm]
+            report_print(f"- team: [{name}]({team_page}), {access}")
+            if name not in team_data:
+                team_data[name] = team
+                # Get the child teams.
+                url = team["url"] + "/teams?per_page=100"
+                team_teams = request_list(url)
+                teams.extend(team_teams)    # Play fast and loose with extending while iterating.
+                team["child_teams"] = team_teams
+
+                # Get the people in the team.
+                url = team["url"] + "/members?per_page=100"
+                team_members = request_list(url)
+                team["members"] = team_members
+
+            for child_team in team_data[name]["child_teams"]:
+                team_name = child_team["name"]
+                team_page = child_team["html_url"]
+                report_print(f"  - team: [{team_name}]({team_page}), {access}")
+
+            for member in team_data[name]["members"]:
+                login = member["login"]
+                if login not in users:
+                    user_info = request_dict(member["url"])
+                    users[login] = user_info
+                user_page = users[login]["html_url"]
+                user_name = users[login]["name"]
+                if user_name is None:
+                    user_name = ""
+                else:
+                    user_name += " "
+                report_print(f"  - {user_name}[@{login}]({user_page})")
+
+    # Get all the teams
+    for team_name, team in team_data.items():
+        screenshot_pages(page, team["html_url"] + "/members", f"team-{file_slug(team_name)}")
+
+    page.close()
+    context.close()
+
+def main():
+    if os.path.exists(IMAGES_DIR):
+        shutil.rmtree(IMAGES_DIR)
+
+    repos = sys.argv[1:]
+    if not repos or repos[0].startswith("-"):
+        print("repo_access_scraper has no options. Just list repos as arguments:")
+        print("  $ repo_access_scraper openedx/ecommerce openedx/xblock")
+        return
+
+    with sync_playwright() as playwright:
+        with open(REPORT_FILE, "w") as report_md:
+            def report_print(*args, **kwargs):
+                print(*args, **kwargs, file=report_md)
+            run(repos, playwright, report_print)
+
+    shutil.make_archive("images", "zip", root_dir=IMAGES_DIR)
+
+    print(f"Browser state, including authentication, is in {DATA_DIR}")
+    print(f"Delete it if you are concerned: rm -rf {DATA_DIR}")
```

Comment:
Is there a reason not to just clean this up everytime?  Does it speed up subsequent runs?

Reply:
If you keep the directory, it means you don't have to log in the next time.

Reply:
I suppose we could flip this: delete the directory unless some gesture to keep it.

Reply:
Actually, I decided to not save the data anywhere. It means you have to login every time you run the tool, but outside of development, this will not be run often.